src/render/three/utils/set-value.ts

```diff
@@ -0,0 +1,62 @@
+import { Object3DNode } from "@react-three/fiber"
+import { Euler, Vector3, Color } from "three"
+import { ThreeRenderState } from "../types"
+
+const setVector =
+    (name: string, defaultValue: number) =>
+    (i: number) =>
+    (instance: Object3DNode<any, any>, value: number) => {
```

Comment:
I wonder how to expand this stuff into other nodes like material nodes and geometry nodes?

To animate color at least we'd want [`MaterialNode`](https://github.com/pmndrs/react-three-fiber/blob/master/packages/fiber/src/three-types.ts#L62) for `motion.meshBasicMaterial` and whatnot

another concept R3F has is "piercing" so you can do like `<mesh material-color="pink" />` maybe we can do something there?

Reply:
This will already animate color, its just typed this way, and supports piercing for transforms. But I think we'd have to do something extra for `material-color`